lib/msgServer/mmrp/index.js

```diff
@@ -103,11 +98,33 @@ function MmrpNode(role, cfg, clusterId) {
 	if (this.isRelay) {
 		this.router = createRouter(cfg);
 		var routerUri = this.router.getsockopt(zmq.ZMQ_LAST_ENDPOINT);
+
 		if (routerUri) {
 			this.routerPort = parseInt(routerUri.substr(routerUri.lastIndexOf(':') + 1), 10);
+
+			// Share MMRP master port through the environment
+			process.env[ENV_MAGE_MMRP_MASTER_PORT] = this.routerPort;
+
+			// Generate the cluster ID based on the port the master listens on
+			this.clusterId = serviceDiscovery.helpers.generateId(this.routerPort);
+
+			// Set the identity of the router socket to the cluster ID
+			this.router.setsockopt(zmq.ZMQ_IDENTITY, new Buffer(this.clusterId));
```

Comment:
I notice this wasn't done before for the router, was it? Any idea why and how does introducing it here change the behavior?

Reply:
Been two months + so memory is fuzzy. But IIRC, in the past indentity was ephemeral, so setting an explicit identity didn't matter. Now that the ID is consistent across reboots, setting it is necessary to ensure queued messages will find their way to that node post-reboot.